scylla/tests/retries.rs

```diff
@@ -11,14 +11,17 @@ use tracing::info;
 use utils::test_with_3_node_cluster;
 
 use scylla_proxy::{
-    Condition, ProxyError, Reaction, RequestOpcode, RequestReaction, RequestRule, WorkerError,
+    Condition, ProxyError, Reaction, RequestOpcode, RequestReaction, RequestRule, ShardAwareness,
+    WorkerError,
 };
 
 #[tokio::test]
 #[ntest::timeout(5000)]
 async fn speculative_execution_is_fired() {
     const TIMEOUT_PER_REQUEST: Duration = Duration::from_millis(1000);
-    let res = test_with_3_node_cluster(217, |proxy_uris, translation_map, mut running_proxy| async move {
+
+    // We choose Unaware to support testing Cassandra
```

Comment:
Why does the proxy panic with Cassandra in query mode? Wouldn't it make more sense in this case to switch to the "unaware" mode?

Reply:
IIUC, you suggest that once a proxy detects a non-shard-aware cluster (effectively a cluster without shards, most probably Cassandra), it should switch from `QueryNode` to `Unaware` ?
I didn't do it so in order to have explicit errors. I can change that, though, but only if I log a warning about that.

Reply:
I think the semantic "just do what the underlying node does" makes more sense. As you found out, the current semantic of `QueryNode` is not useful with Cassandra clusters, and I'm not sure if I see a use for the current semantic - maybe you do, and if so then maybe those two semantics could live in parallel as separate modes?